app/models.py

```diff
@@ -241,7 +241,8 @@ class ServicePermission(db.Model):
     service = db.relationship("Service")
     created_at = db.Column(db.DateTime, default=datetime.datetime.utcnow, nullable=False)
 
-    service_permission_types = db.relationship(Service, backref=db.backref("permissions"))
+    service_permission_types = db.relationship(
+        Service, backref=db.backref("permissions", cascade="all, delete-orphan"))
```

Comment:
Need the (cascade)[http://docs.sqlalchemy.org/en/latest/orm/cascades.html] for it to work with association_proxy, if the concern is that the permission_service_types permission is removed if there are no parents attached to it, then it doesn't do this and I have a test to ensure that this is the case - test_remove_service_does_not_remove_service_permission_types